gr-filter/lib/ival_decimator_impl.cc

```diff
@@ -0,0 +1,98 @@
+/* -*- c++ -*- */
+/*
+ * Copyright 2020 Free Software Foundation, Inc.
+ *
+ * This file is part of GNU Radio
+ *
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ *
+ */
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include "ival_decimator_impl.h"
+#include <gnuradio/io_signature.h>
+
+namespace gr {
+namespace filter {
+
+// Convenience defines for work function
+#define DT_IVALDECIM_BYTE 1
+#define DT_IVALDECIM_SHORT 2
+
+ival_decimator::sptr ival_decimator::make(int decimation, int data_type, int data_size)
+{
+    return gnuradio::get_initial_sptr(
+        new ival_decimator_impl(decimation, data_type, data_size));
+}
+
+/*
+ * The private constructor
+ */
+ival_decimator_impl::ival_decimator_impl(int decimation, int data_type, int data_size)
+    : gr::sync_decimator("ichar_decimator",
+                         gr::io_signature::make(1, 1, data_size),
+                         gr::io_signature::make(1, 1, data_size),
+                         decimation),
+      d_data_type(data_type)
+{
+    d_doubledecimation = 2 * decimation;
+
+    if ((d_data_type < 1) || d_data_type > 2)
```

Comment:
need a check for `data_size` too

Reply:
can you just use `data_type` for both it and `data_size` ... since they happen to overlap exactly?